src/api/Metadata.js

```diff
@@ -1053,6 +1054,82 @@ class Metadata extends File {
 
         return getProp(suggestionsResponse, 'data.suggestions', []);
     }
+
+    /**
+     * Build URL for metadata options associated to a taxonomy field.
+     *
+     * @param scope
+     * @param templateKey
+     * @param fieldKey
+     * @returns {`${string}/metadata_templates/${string}/${string}/fields/${string}/options`}
+     */
+    getMetadataOptionsUrl(scope: string, templateKey: string, fieldKey: string): string {
+        return `${this.getBaseApiUrl()}/metadata_templates/${scope}/${templateKey}/fields/${fieldKey}/options`;
+    }
+
+    /**
+     * Gets options associated with a taxonomy field.
+     *
+     * @param id
+     * @param scope
+     * @param templateKey
+     * @param fieldKey
+     * @param level
+     * @param options
+     * @returns {Promise<MetadataOptions>}
+     */
+    async getMetadataOptions(
+        id: string,
+        scope: string,
+        templateKey: string,
+        fieldKey: string,
+        level: number,
+        options: { marker?: string, searchInput?: string, signal?: AbortSignal },
+    ) {
+        this.errorCode = ERROR_CODE_FETCH_METADATA_OPTIONS;
+
+        if (!id) {
+            throw getBadItemError();
+        }
+
+        if (!scope) {
+            throw new Error('Missing scope');
+        }
+
+        if (!templateKey) {
+            throw new Error('Missing templateKey');
+        }
+
+        if (!fieldKey) {
+            throw new Error('Missing fieldKey');
+        }
+
+        // 0 is a valid level value
+        if (!level && level !== 0) {
+            throw new Error('Missing level');
+        }
+
+        const url = this.getMetadataOptionsUrl(scope, templateKey, fieldKey);
+        const { marker, searchInput, signal } = options;
+        const params = {
+            ...(marker ? { marker } : {}),
+            ...(searchInput ? { searchInput } : {}),
+        };
+
+        if (signal?.aborted) {
+            this.xhr.abort();
+
+            return {};
+        }
+
+        const metadataOptions = this.xhr.get({
+            url,
+            id: getTypedFileId(id),
+            params,
+        });
+
+        return getProp(metadataOptions, 'data', {});
```

Comment:
I think there has to be some transformation of the response here. The fetcher requires a response of [this](https://git.dev.box.net/Box/frontend-mono/blob/master/packages/group-shared-features/metadata-editor/src/lib/components/combobox-with-api-pagination/types.ts#L12-L15) format. Where `T` in our case is[ this](https://git.dev.box.net/Box/frontend-mono/blob/master/packages/group-shared-features/metadata-editor/src/lib/components/metadata-editor-fields/components/metadata-taxonomy-field/types.ts#L18-L24).

This is what I have in my branch:
```export type TaxonomyOptionsFetcher = (
    id: string,
    templateKey: string,
    scope: string,
    fieldKey: string,
    level: number,
    options: PaginationQueryInput,
) => Promise<FetcherResponse<TaxonomyComboboxOption>>;```